pyop2/types/dat.py

```diff
@@ -492,6 +492,39 @@ def norm(self):
         from math import sqrt
         return sqrt(self.inner(self).real)
 
+    def maxpy(self, scalar: list, x: list) -> None:
```

Comment:
```suggestion
    def maxpy(self, scalar: Sequence, x: Sequence) -> None:
```
this probably entails a `from collections.abc import Sequence` at the top of the file.